### PR TITLE
show user to whom you are sending clout on the send-bitclout page

### PR DESCRIPTION
**Pull request by lazynina**
Use search-bar component to get username and avatar when searching for user on Send BitClout page.

image

image

Updated search bar component to search for single profile based on public key if search text looks like a public key.

Supports public keys with a profile:
image

### CODE REVIEW

File: src/app/search-bar/search-bar.component.ts [hunk 1]
```diff
@@ -45,7 +54,44 @@ export class SearchBarComponent {
       readerPubKey = this.globalVars.loggedInUser.PublicKeyBase58Check;
     }
 
-    this.backendApi
+    // If we are searching for a public key, call get single profile with the public key.
+    if (
```
Comment (maebeam) [inline src/app/search-bar/search-bar.component.ts:58]: Do we do these checks anywhere else in the repo? If so, worth making a helper?

**Review by maebeam — APPROVED** [review]
Looks great!

**Comment by maebeam** [comment]
How does this look for blacklisted users?

File: src/app/search-bar/search-bar.component.ts (hunk 1 above)
Comment (lazynina) [reply src/app/search-bar/search-bar.component.ts:58]: we have a function called `isMaybePublicKey` :joy: I'll update it to include length checks and use it here.

**Comment by lazynina** [comment]
> How does this look for blacklisted users?

line 66 in `search-bar.component.ts`, if a user is blacklisted is returned by the get-single-profile call, we don't display that user.  You will not be able to send to blacklisted keys.

The call to `/get-profiles` already filters out blacklisted public keys for us.

**Pull request merged**